source/signaling_api.c

```diff
@@ -0,0 +1,1810 @@
+                    {
+                        if( pair.valueLength >= SIGNALING_CHANNEL_TTL_SECONDS_BUFFER_MAX )
+                        {
+                            /* Unexpected TTL value from cloud. */
+                            result = SIGNALING_RESULT_INVALID_TTL;
+                            break;
+                        }
+
+                        strncpy( ttlSecondsBuffer, pair.value, pair.valueLength );
+                        pChannelInfo->messageTtlSeconds = ( uint32_t ) strtoul( ttlSecondsBuffer, NULL, 10 );
+
+                        if( ( pChannelInfo->messageTtlSeconds < SIGNALING_CHANNEL_TTL_SECONDS_MIN ) ||
+                            ( pChannelInfo->messageTtlSeconds > SIGNALING_CHANNEL_TTL_SECONDS_MAX ) )
+                        {
+                            /* Unexpected TTL value from cloud. */
+                            result = SIGNALING_RESULT_INVALID_TTL;
+                        }
+                    }
+                    else
+                    {
+                        /* Unknown attribute. */
+                        result = SIGNALING_RESULT_INVALID_JSON;
+                    }
+                }
+                else
+                {
+                    /* Invalid single master configuration. */
+                    result = SIGNALING_RESULT_INVALID_JSON;
+                }
+            }
+            else if( strncmp( pair.key, "Version", pair.keyLength ) == 0 )
+            {
+                pChannelInfo->pVersion = pair.value;
+                pChannelInfo->versionLength = pair.valueLength;
+            }
+            else {
+                /* Skip unknown attributes. */
+            }
+
+            if( result != SIGNALING_RESULT_OK )
+            {
+                break;
+            }
+
+            jsonResult = JSON_Iterate( pChannelInfoBuffer, channelInfoBufferLength, &( channelInfoStart ), &( channelInfoNext ), &( pair ) );
+        }
+    }
+
+    return result;
+}
+
+/*-----------------------------------------------------------*/
+
+SignalingResult_t Signaling_ConstructDescribeMediaStorageConfigRequest( SignalingAwsRegion_t * pAwsRegion,
+                                                                        SignalingChannelArn_t * pChannelArn,
+                                                                        SignalingRequest_t * pRequestBuffer )
+{
+    SignalingResult_t result = SIGNALING_RESULT_OK;
+    int snprintfRetVal = 0;
+
+    if( ( pAwsRegion == NULL ) ||
+        ( pAwsRegion->pAwsRegion == NULL ) ||
+        ( pChannelArn == NULL ) ||
+        ( pRequestBuffer == NULL ) ||
+        ( pRequestBuffer->pUrl == NULL ) ||
+        ( pRequestBuffer->pBody == NULL ) ||
+        ( pChannelArn->pChannelArn == NULL ) )
+    {
+        result = SIGNALING_RESULT_BAD_PARAM;
+    }
+
+    if( result == SIGNALING_RESULT_OK )
+    {
+        if( SIGNALING_IS_CHINA_REGION( pAwsRegion ) )
+        {
+            snprintfRetVal = snprintf( pRequestBuffer->pUrl,

```

Comment:
Should add other cases?

https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/pull/430/files

Reply:
Again, Could you help me understand if it is necessary to apply that into WebRTC? In my understanding, we're not using curl in WebRTC flow.

Reply:
Not on the current roadmap so not needed in this pr